app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ActiveRecord::Base
+  has_secure_password
+
+  has_many :groups, foreign_key: :creator_id
+  has_many :snippets, through: :groups
+
+  validates :email, presence: true, uniqueness: true
+  validates :username, presence: true, uniqueness: true
+
+  before_save :downcase_email
+
+  private
+
+  def downcase_email
+    self.email = email.downcase
```

Comment:
Hi, I'd avoid this (I know I did it the other day but I shouldn't have.) Rather, make the uniqueness constraint  case_sensitive: false so that users can keep their emails in the case they prefer but you still keep the validation you want.